extensions/amp-story-captions/0.1/amp-story-captions.js

```diff
@@ -46,6 +46,9 @@ export class AmpStoryCaptions extends AMP.BaseElement {
     } else {
       this.element.appendChild(this.container_);
     }
+    if (this.element.hasAttribute('auto-append')) {
+      this.container_.classList.add(`amp-story-captions-auto-append`);
```

Comment:
I think you could set a class direclty from the JSX template

Reply:
The fact that `style-preset=default` behaves different depending on if we automatically appended it or not is tripping me a bit.

Can you provide an example of how you position the `default` preset as a publisher, and what the default position is if a publisher doesn't write any code?

Reply:
It trips me a bit too.
I think the thought process is this:
If it's auto-appended we base it on the expected UX of captions and position it relative to the video.
It it's publisher specified the assumption is that it's intentionally placed.

The example for publisher writing no code is already in [examples/amp-story/amp-story-captions.html](https://github.com/ampproject/amphtml/pull/37967/files#diff-94f5bebdf0bb21b91c0ea7c174e04f99ae0ddf966cb8b9e3a6bb4b3ab72fe8f9) as part of #37967

Reply:
Okay it's a bit tricky, thank you for sharing.
Two things would be awesome if we could make them work together:
1. Using `style-preset=default` works awesome out of the box, without requiring a single line of CSS
2. Publishers can override the default position set by `style-preset=default`

If both things can't coexist, then let's keep things how you have them.

Reply:
To follow up from our discussion:
1. It looks awesome out of the box both inline and auto-attached.
2. It can be placed anywhere when inline. When auto-attached it is locked in a position.